torch_xla/csrc/convolution.h

```diff
@@ -1,11 +1,252 @@
 #ifndef XLA_TORCH_XLA_CSRC_CONVOLUTION_H_
 #define XLA_TORCH_XLA_CSRC_CONVOLUTION_H_
 
+#include <string>
+
 #include "absl/types/span.h"
 #include "tensorflow/compiler/xla/client/xla_builder.h"
+#include "tensorflow/compiler/xla/xla_data.pb.h"
+#include "tensorflow/compiler/xla/client/xla_builder.h"
+#include "tensorflow/tsl/lib/gtl/inlined_vector.h"
+#include "tensorflow/tsl/platform/stringpiece.h"
+#include "tensorflow/tsl/platform/errors.h"
 
 namespace torch_xla {
 
+// Tensor format for input/output activations used in convolution operations.
+// The mnemonics specify the meaning of each tensor dimension sorted from
+// largest to smallest memory stride.
+// N = Batch, H = Image Height, W = Image Width, C = Number of Channels.
+// This code copyied from TF: 
+// https://github.com/tensorflow/tensorflow/blob/7f47eaf439d2b81de1aa24b10ed57eabd519dbdb/tensorflow/core/util/tensor_format.h#L38
+enum PTXLATensorFormat {
```

Comment:
ditto, put these in a separate helper files.

Reply:
Thanks, Jack, updated